core/payment/src/dao/invoice.rs

```diff
@@ -176,6 +179,33 @@ impl<'c> InvoiceDao<'c> {
         .await
     }
 
+    pub async fn last_invoice_stats(
+        &self,
+        node_id: NodeId,
+        since: DateTime<Utc>,
+    ) -> DbResult<BTreeMap<(Role, DocumentStatus), StatValue>> {
+        let results = readonly_transaction(self.pool, move |conn| {
+            let invoices: Vec<ReadObj> = query!()
+                .filter(dsl::owner_id.eq(node_id))
+                .filter(dsl::timestamp.gt(since.naive_utc()))
+                .load(conn)?;
+            Ok::<_, DbError>(invoices)
+        })
+        .await?;
+        let mut stats = BTreeMap::<(Role, DocumentStatus), StatValue>::new();
+        for invoice in results {
+            let key = (invoice.role, DocumentStatus::try_from(invoice.status)?);
+            let entry = stats.entry(key).or_default();
+            let total_amount = invoice.amount.0;
+            *entry = entry.clone()
```

Comment:
Can't we use `+=` here instead of cloning since `StatValue` implements `AddAssign`?